### ISSUE TITLE
to_datetime followed by update on column ruins data type and formatting

### ISSUE THREAD

**Issue by ctivanovich** [pandas-dev/pandas#23054]
Before transformation of birthday entries stupidly recorded as floating-point values by company partner:
```
0            NaN
1    1.98508e+07
2    1.98811e+07
3            NaN
4            NaN
5            NaN
6            NaN
7            NaN
8     1.9951e+07
9    1.99605e+07

```
After I apply the following code to non-null values:
```python
In:
pd.to_datetime(df.birthday[~df.birthday.isnull()].astype(int).astype(str), format="%Y%m%d")

Out:
1         1985-08-11
2         1988-11-07
8         1995-10-01
9         1996-05-03
15        1986-05-19
16        1996-06-19
17        1995-06-18
27        1994-04-01
29        1997-08-22
30        1990-08-04
```

Great, exactly what I want. But, when I go to update the dataframe with these lovely date values:
`df_backup.birthday.update(pd.to_datetime(df.birthday[~df.birthday.isnull()].astype(int).astype(str), format="%Y%m%d"))`, the birthday values in the dataframe become:

```
0                         NaN
1          492566400000000000
2          594864000000000000
3                         NaN
4                         NaN
5                         NaN
6                         NaN
7                         NaN
8          812505600000000000
9          831081600000000000
```

It occurred to me that this may be happening because the column I'm updating is a float-type column, and so the datetime objects are awkwardly coerced to floats, but then, how to allow for the presence of NaNs and datetime objects in this column? I tried first changing the birthday column to type object before updating with datetime objects, but that didn't solve the issue. I will try the workaround of left joining as a new column.

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.6.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 61 Stepping 4, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.23.4
pytest: 3.8.2
pip: 18.0
setuptools: 40.4.3
Cython: 0.28.5
numpy: 1.15.2
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 7.0.1
sphinx: 1.8.1
patsy: 0.5.0
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: 1.2.1
tables: 3.4.4
numexpr: 2.6.8
feather: None
matplotlib: 3.0.0
openpyxl: 2.5.8
xlrd: None
xlwt: None
xlsxwriter: 1.1.1
lxml: 4.2.5
bs4: 4.6.3
html5lib: 1.0.1
sqlalchemy: 1.2.12
pymysql: None
psycopg2: 2.7.5 (dt dec pq3 ext lo64)
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

**Comment by jorisvandenbossche** [comment]
Can you provide a small reproducible example?

**Comment by ctivanovich** [comment]
So weird, I can't reproduce it with a toy example, but it definitely happens on my full data set. The data set I have to work with also had a few hand-entered "No data" values that I dropped by string type and replaced with np.nan before doing the datetime conversion, maybe that is related? I also included this aspect in my recreation attempt, but, it's not causing problems. Maybe there is some other small detail to this that I'm missing. 

Just attaching the cleaned dates as a new column and dropping the ugly one worked though, so at least I can push forward at work.
